ui/src/main/js/MainPage.js

```diff
@@ -79,12 +80,14 @@ class MainPage extends Component {
             );
         }
 
+        const renderComponent = (name === 'channel_slack') ? <SlackGlobalConfiguration /> : <GlobalConfiguration key={name} descriptor={component} />;
```

Comment:
Can we create a map of _name_ to static component and then loop over that?

Reply:
We could perhaps use DESCRIPTOR_NAME.CHANNEL_SLACK from the descriptorUtilites.js file. That defines the constants today.  We can either refactor that class to only contain the descriptor names and operations definitions.  Or create a new js file to start from scratch.  Have a look at descriptorUtilities.js and decide from there.

Reply:
I introduced a new model For Slack info that would hold this data as I want to separate as much of the implementation from the descriptors as possible.

Reply:
As to your recommendation Gavin, I did debate doing it that way. I'll push a change for your viewing pleasure

Reply:
Upon further investigation we may want to stick with this approach for now. The other configurations will require props to be passed when they're implemented (we'll need the if statement anyway to see what props to pass) and this is also temporary until all the dynamic items are removed.